xapian-letor/ranker/ranker.cc

```diff
@@ -234,7 +234,8 @@ initialise_queryparser(const Xapian::Database & db)
     parser.set_default_op(Xapian::Query::OP_OR);
     parser.set_stemmer(stemmer);
     parser.set_stemming_strategy(Xapian::QueryParser::STEM_SOME);
-    parser.set_stopper(&mystopper);
+    parser.set_stopper(mystopper);
+    mystopper->release();
```

Comment:
The docs say:

> You can hand ownership of a dynamically allocated `Stopper` object to Xapian by calling `release()` and then passing the object to a Xapian method.

However, you do things the other way around - i.e. pass the object to a Xapian method and **then** call `release()` on this.  

Your order doesn't work correctly - the object is now being reference counted, but nothing owns it (because it wasn't being reference counted when you passing it to the method, so that didn't add a reference).  In this case that probably means it will never be released (i.e. a memory leak), though it could also mean it got released before being done with (if `QueryParser` now passes it to something else which temporarily takes a reference and releases it again).

`release()` actually returns the object it was called on, so you can release and hand-over in one:

```
parser.set_stopper(mystopper->release());
```